From 60bd6a2e2b15418c91b5a962c5656ef740d34ae2 Mon Sep 17 00:00:00 2001
From: Radha Mohan Chintakuntla <radhac@marvell.com>
Date: Mon, 12 Apr 2021 01:58:53 -0700
Subject: [PATCH 1418/1921] octeontx2-sdp: assign VF0 rings as SDP rings per PF

We use SDP VF0 in EP side as PF on the host. This enables us to attach
the SDP VF0 to a dataplane application and the PF can handle FLR.

Signed-off-by: Radha Mohan Chintakuntla <radhac@marvell.com>
Change-Id: Ic18041485865858adc7a236b751ca80f7bc013f1
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/49702
Reviewed-by: Satananda Burla <sburla@marvell.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/soc/marvell/octeontx2-sdp/sdp.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2-sdp/sdp.c b/drivers/soc/marvell/octeontx2-sdp/sdp.c
index 152da151a960..832f782e4ee8 100644
--- a/drivers/soc/marvell/octeontx2-sdp/sdp.c
+++ b/drivers/soc/marvell/octeontx2-sdp/sdp.c
@@ -1258,8 +1258,9 @@ static int sdp_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 
 	fw_rinfo.u = 0;
 	fw_rinfo.s.dir = FW_TO_HOST;
-	fw_rinfo.s.rppf = info.num_pf_rings;
-	fw_rinfo.s.rpvf = info.vf_rings[0];
+	/* PF doesn't have any rings */
+	fw_rinfo.s.rppf = info.vf_rings[0];
+	fw_rinfo.s.rpvf = info.vf_rings[1];
 	fw_rinfo.s.numvf = info.max_vfs - 1;
 	fw_rinfo.s.pf_srn = info.pf_srn;
 
-- 
2.31.1

